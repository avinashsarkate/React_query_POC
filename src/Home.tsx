import React from 'react'
import Navbar from './Navbar/Navbar'
import { Notes } from './Notes'


const Home = () => {
  return (
    <div>
        <Navbar/>
       <Notes/>
    </div>
  )
}

export default Home