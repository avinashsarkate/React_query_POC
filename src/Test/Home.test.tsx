

/* eslint-disable testing-library/prefer-screen-queries */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import Home from "../Home";


jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: {} })),
 delete:jest.fn(() => Promise.resolve({ data: {} })),
 post: jest.fn(() => Promise.resolve({ data: {} }))
}))


describe("Home Component", () => {

  const queryClient = new QueryClient();

  const component = (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </QueryClientProvider>
  );




  test("Render component without crashing", () => {
    render(component);
  });
 ;
  test("Render component", () => {
    const { container } = render(component);
    expect(container).toBeTruthy();
  });

  test("to create snapshot", () => {
    const { asFragment } = render(component);
    expect(asFragment).toMatchSnapshot();
  });
});

