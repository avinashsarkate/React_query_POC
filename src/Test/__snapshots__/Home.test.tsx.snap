// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component to create snapshot 1`] = `[Function]`;
